smithy-model/src/main/java/software/amazon/smithy/model/knowledge/HttpBindingIndex.java

```diff
@@ -368,15 +368,13 @@ private String determineContentType(
             }
 
             if (binding.getLocation() == HttpBinding.Location.PAYLOAD) {
-                if (binding.getMember().hasTrait(EventStreamTrait.class)) {
-                    return eventStreamContentType;
-                }
-
                 Shape target = model.getShape(binding.getMember().getTarget()).orElse(null);
                 if (target == null) {
                     // Can't determine the content-type because the model is broken :(
                     // Let other parts of the validation system point this out.
                     break;
+                } else if (target.isUnionShape() && target.hasTrait(StreamingTrait.class)) {
```

Comment:
Should we add an `isEventStream` method somewhere to centralize this behavior? Like on the EventStreamTrait as a static method?

Reply:
sure